src/mono/mono/mini/debugger-engine.c

```diff
@@ -1323,8 +1323,11 @@ mono_de_ss_start (SingleStepReq *ss_req, SingleStepArgs *ss_args)
 	} else {
 		frame_index = 1;
 
+#ifndef TARGET_WASM
 		if (ss_args->ctx && !frames) {
-
+#else
+		if (!frames) {
```

Comment:
Why was this causing the issue with the test? Why did it specifically break the async test?

Reply:
Because once it doesn't execute `ss_calculate_framecount` because `ss_args->ctx == NULL`, `nframes` variable would be 0, and wouldn't enter in this if: `if (asyncMethod && asyncMethod->num_awaits && nframes && rt_callbacks.ensure_jit (frames [0])) {` and wouldn't call `ss_bp_add_one`

Reply:
Okay. `nframes ==0` would also cause the loop starting ` line 1341 to get skipped, IIUC. Is there any case where that loop will get incorrectly skipped? `catch`(not the last one, IIUC) in an async method? Or catch in non-async methods?

Just trying to understand with silly/naive questions 😬

Reply:
I think this loop would be skipped wrongly without the fix. Now the value of `nframes` and `frames` variables will be correct for both loops, line 1341 and line 1358.